tls-client/main.cpp

```diff
@@ -224,15 +219,25 @@ class HelloHTTPS {
                 ret == MBEDTLS_ERR_SSL_WANT_WRITE));
         if (ret < 0) {
             print_mbedtls_error("mbedtls_ssl_handshake", ret);
+            _tcpsocket->close();
             return;
         }
 
+        /* Fill the request buffer */
+        _bpos = snprintf(_buffer, sizeof(_buffer) - 1, "GET %s HTTP/1.1\nHost: %s\n\n", path, HTTPS_SERVER_NAME);
+        
+        int written = 0;
         do {
             ret = mbedtls_ssl_write(&_ssl, (const unsigned char *) _buffer, _bpos);
```

Comment:
The [manual for this function](https://tls.mbed.org/api/ssl_8h.html) says that it has to be called with updated arguments in case the write was partial.